pkg/daemon/controller/nodeinfo.go

```diff
@@ -324,19 +324,50 @@ func ensureVxlanInterfaceAddresses(vxlanDev *vxlan.Device, addresses []netlink.A
 	return nil
 }
 
-func checkNodeUpdate(updateEvent event.UpdateEvent) bool {
-	oldNode := updateEvent.ObjectOld.(*corev1.Node)
-	newNode := updateEvent.ObjectNew.(*corev1.Node)
+// createOrUpdateIPInstance will create or update an NodeInfo
+func (r *nodeInfoReconciler) createOrUpdateNodeVxlanInfo(thisNode *corev1.Node,
+	vtepIP net.IP, vtepMac net.HardwareAddr, nodeLocalVxlanAddr []netlink.Addr) (info *networkingv1.NodeInfo, err error) {
+	var nodeInfo = &networkingv1.NodeInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.ctrlHubRef.config.NodeName,
+		},
+	}
+
+	_, err = controllerutil.CreateOrPatch(context.TODO(), r, nodeInfo, func() error {
+		if !nodeInfo.DeletionTimestamp.IsZero() {
+			return fmt.Errorf("node info %s is deleting, can not be updated", nodeInfo.Name)
+		}
+
+		nodeInfo.Spec.VTEPInfo = &networkingv1.VTEPInfo{
+			IP:       vtepIP.String(),
+			MAC:      vtepMac.String(),
+			LocalIPs: utils.GenerateIPStringList(nodeLocalVxlanAddr),
+		}
+
+		nodeInfo.OwnerReferences = []metav1.OwnerReference{
+			*ipamutils.NewControllerRef(thisNode, corev1.SchemeGroupVersion.WithKind(nodeKind),
+				true, false),
+		}
 
-	if oldNode.Annotations[constants.AnnotationNodeVtepIP] != newNode.Annotations[constants.AnnotationNodeVtepIP] ||
-		oldNode.Annotations[constants.AnnotationNodeVtepMac] != newNode.Annotations[constants.AnnotationNodeVtepMac] ||
-		oldNode.Annotations[constants.AnnotationNodeLocalVxlanIPList] != newNode.Annotations[constants.AnnotationNodeLocalVxlanIPList] {
-		return true
+		return nil
+	})
+
+	var updateTimestamp string
```

Comment:
if `CreateOrPatch` return `OperationNone` (aka, no-op), then timestamp should not be updated

Reply:
fixed